### ISSUE TITLE
HDMI Audio keeps cutting

### ISSUE THREAD

**Issue by JacksonEpworth** [raveberry/raveberry#104]
The HDMI audio keeps cutting in and out. The HDMI audio is normal when playing other audio sources, but with Raveberry it keeps cutting in and out. With Analogue audio it works fine

**Comment by raveberry** [comment]
What hardware are you using? Did you enable `screen_visualization` during setup?

**Comment by JacksonEpworth** [comment]
I am using the Pi4 and I did enable screen_visualization during setup

**Comment by raveberry** [comment]
Cool to see someone use that feature!
For screen visualization, audio is put into a buffer before playing in order to synchronize it with the visualization. This buffer is pretty large and I assume that it is the cause for this problem.
I experienced intermittent playback before as well, but not enough to make me investigate.

Maybe there is another solution for the delay problem, but in the meantime you can edit `/etc/mopidy/mopidy.conf` and remove the delay. Change
`output = rgvolume ! audioconvert ! audioresample ! tee name=t ! queue ! pulsesink server=127.0.0.1 device=cava t. ! queue max-size-bytes=85000000 max-size-buffers=0 max-size-time=0 ! pulsesink server=127.0.0.1 ts-offset=85000000`
to
`output = rgvolume ! audioconvert ! audioresample ! tee name=t ! queue ! pulsesink server=127.0.0.1 device=cava t. ! pulsesink server=127.0.0.1`

This should hopefully fix audio playback issues. This is

**Comment by JacksonEpworth** [comment]
I made the changes to the config and rebooted the Pi, I am still experience the issue

**Comment by raveberry** [comment]
hm, ok. Then I'm a little clueless to be honest.
I don't know why the audio would cut only for hdmi output only with Raveberry. How bad is it? Does it also occur with
`PULSE_SERVER=127.0.0.1 paplay /usr/share/sounds/alsa/Front_Center.wav`
?

The other thing that comes to my mind would be to remove the second sink altogether (though the visualization won't work anymore):
`output = rgvolume ! audioconvert ! audioresample ! pulsesink server=127.0.0.1`

**Comment by JacksonEpworth** [comment]
How bad is it? It cuts out every second for half a second
It doesn't occur with the command 
Here's an example: https://streamable.com/hj2ap1

The example was taken after the new config change

**Comment by raveberry** [comment]
Ok, this is way worse than what I experienced. I don't know what else to do instead of crosstesting where the issue is coming from (raveberry, pulseaudio, mopidy, hdmi or something else).
I will try to test it myself the next few days and will report back.

**Comment by raveberry** [comment]
tl;dr: edit /etc/pulse/system.pa and change
`load-module module-udev-detect`
to
`load-module module-udev-detect tsched=0`

With strong enough google-fu, this issue is quickly solved:
https://wiki.archlinux.org/index.php/PulseAudio/Troubleshooting#Glitches.2C_skips_or_crackling

Unfortunately mine wasn't strong enough, so I had to dig for the cause of this problem. Testing with different players (mplayer, omxplayer) and different formats (wav, m4a, mp3) yielded that the audio format does not matter at all, what does matter is whether pulse is used for output.
The next realization was that the issue occured depending on whether the system service was used or not. After eliminating command line arguments the only thing left to check was the config files. Comparing these two files, I found that the user mode pulse does have the `tsched=0` by default, but the system one does not.

Apparently, timer-based audio scheduling breaks on the Raspberry Pi with hdmi and we have to disable it. The next update will include a corresponding change, until then you can manually edit the file. You can reset your mopidy config by pressing "Disable Streaming" in `/settings` and everything should work fine again.

**Comment by JacksonEpworth** [comment]
Awesome, thank you so much.
